### PR TITLE
Bmcdole/de41139/handle add edit failure

### PR DESCRIPTION
**Pull request by bamcdole**

### CODE REVIEW

File: src/api/scheduler.js [hunk 1]
```diff
@@ -4,43 +4,53 @@ export class Scheduler {
...
+	static async setEnable(scheduleId, isEnabled) {
+		return await this._put(Routes.SetEnable(scheduleId, isEnabled));
 	}
 
 	// Helper Methods
 
+	static async _fetch(url, options) {
+		return await fetch(url, options)
+			.then(response => {
+				if (!response.ok) {
```
Comment (bamcdole) [inline src/api/scheduler.js:44]: The 'fetch' command doesn't throw on certain error return codes (ex: 500). Instead, it denotes the failure using this 'ok' property. This wrapper method gives us the expected behaviour: "Throw on error"

File: lang/en.js [hunk 2]
```diff
@@ -25,6 +25,7 @@ export default {
 	'add.SelectDataSet': 'Select Your Advanced Data Set',
 	'add.ConfigureSchedule': 'Configure Your Schedule',
 	'add.DeliveryMethod': 'Choose Your Delivery Method',
+	'add.FailedToSave': 'Failed to save schedule',
```
Comment (bamcdole) [inline lang/en.js:28]: Bare minimum feedback. Will need to consult Josh, to see if we should provide a more diagnostic message.

File: src/api/scheduler.js [hunk 3]
```diff
@@ -4,43 +4,53 @@ export class Scheduler {
```
Comment (bamcdole) [inline src/api/scheduler.js:4]: These methods are async

File: src/components/d2l-wizard-manager.js [hunk 4]
```diff
@@ -161,7 +161,13 @@ class WizardManager extends LocalizeMixin(LitElement) {
 
 	async _handleStepThreeDone() {
 		if (this.deliveryMethod.validate()) {
-			await this._saveSchedule();
+			try {
+				await this._saveSchedule();
+			} catch (e) {
+				window.console.error(e);
```
Comment (IhabAlkabary) [inline src/components/d2l-wizard-manager.js:167]: this can go

File: src/components/d2l-wizard-manager.js [hunk 5]
```diff
@@ -161,7 +161,13 @@ class WizardManager extends LocalizeMixin(LitElement) {
...
+				this.shadowRoot.getElementById('failed-to-save').setAttribute('open', '');
```
Comment (IhabAlkabary) [inline src/components/d2l-wizard-manager.js:168]: Can you share a printscreen showing how this renders

**Review by lweber-d2l — COMMENTED** [review]
Looks good, just waiting for a response to Ihab's questions.

File: src/components/d2l-wizard-manager.js (hunk 4 above)
Comment (bamcdole) [reply src/components/d2l-wizard-manager.js:167]: The console log you mean?

Comment (IhabAlkabary) [reply src/components/d2l-wizard-manager.js:167]: yeah

File: src/components/d2l-wizard-manager.js (hunk 5 above)
Comment (bamcdole) [reply src/components/d2l-wizard-manager.js:168]: image

**Review by lweber-d2l — APPROVED** [review]
Screenshot looks good 👍

**Review by IhabAlkabary — DISMISSED** [review]

**Pull request merged**